### DISCUSSION TITLE
Adding new functionalities to a Remote Control

### DISCUSSION THREAD

**Discussion by vBruegge** [bitcraze/discussions#984] [Q&A] [answered]
Hi guys,
I'm currently working on the flapper drone and wanted to add some new features. For this, I want to add a new functionality to the remote control. Until now, just the sticks are working, so I thought about using the toggle switches in the back of the RC. Is it possible to add your own code? Where do I find the corresponding code and how do I get access to the new signal?
Thanks in advance,
Lina

**Comment by krichardsson** [comment]
@matejkarasek Do you have any input on this?

**Comment by matejkarasek** [comment, marked as answer]
Hi! 
While this question concerns our Flapper drone in particular, it is actually applicable also to other Crazyflie platforms.

So just to set the context for others:
The remote we deliver uses Edge TX firmware and a 4-in-1 multimodule, which can connect directly to the RFX2411N radio on the crazyflie. We have made some further updates to the multimodule firmware that enable transmitting additional channels (next to roll, pitch, yaw and throttle), the code is here:
https://github.com/flapper-drones/DIY-Multiprotocol-TX-Module/tree/CFlie_improved
And you can also get the compiled firmware file here (this is only meant for 4-in-1 multimodules with STM):
https://flapper-drones.com/docs/doku.php?id=nimbleplus:flight-controller#radio_transmitter_with_a_multimodule

@vBruegge All you need to do is add additional channels to the model inside the transmitter and map them to the switches/toggles, this is done on the Mixer page like this:
https://edgetx.gitbook.io/edgetx-user-manual/b-and-w-radios/model-select/inputs-mixes-and-outputs/mixes

The 1st auxiliary channel is used as a switch between attitude and rate stabilization, and you can either modify this or add code to readout more aux channels:
https://github.com/bitcraze/crazyflie-firmware/blob/2733505868581de8c4a7201ab555dadc2493c84a/src/modules/src/crtp_commander_generic.c#L228

**Reply by vBruegge** [reply]
Thanks a lot for the quick reply, I will try it and will otherwise come back to you!
